### ISSUE TITLE
False positive inconsistent-return-statements with NoReturn function

### ISSUE THREAD

**Issue by jamesbraza** [PyCQA/pylint#4315]
### Steps to reproduce

Given a file `a.py`:
```python
# pylint: disable=missing-docstring

from typing import NoReturn

def no_return(exc: Exception) -> NoReturn:
    """This function doesn't return, as shown by type hint of NoReturn."""
    raise Exception("Blows up") from exc

def return_int() -> int:
    """Function that doesn't return upon ValueError."""
    try:
        return 5
    except ValueError as exc:
        no_return(exc)
```

### Current behavior

Result of `pylint a.py`:
```
a.py:9:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
```

### Expected behavior

Since there is not a return during the `except` case, this pylint error is a false positive.

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.4
astroid 2.5.2
Python 3.8.5 (default, Oct 26 2020, 18:36:15)
[Clang 12.0.0 (clang-1200.0.32.27)]
```

**Comment by cdce8p** [comment]
This functionality was added yesterday, see #4304 😃 
Just for reference, duplicate of #4122

**Comment by jamesbraza** [comment]
Wow this is a blatant duplicate with quite coincidental timing.  Glad it's resolved.
